### ISSUE TITLE
Question about requiring __init__ docstrings

### ISSUE THREAD

**Issue by marc2982** [PyCQA/pydocstyle#43]
I can't seem to find any good examples on the internet concerning meaningful `__init__` docstrings. There may be times when there are no parameters, or perhaps documenting the parameters would be helpful.

PEP 257 does say:

> Public methods (including the `__init__` constructor) should also have docstrings.

but applying this rule seems unnecessary at times.

This is what makes sense to me:

``` python
class Klass:

    """Docstring here."""

    def __init__(self, component):
        """

        component - The component to model. Can be a name or something else.

         """
         self.component = component
```

But of course, pep257 gives this:

> PEP257 Short description should end with a period.

This is what I have ended up doing, but it seems awkward and unnecessary just to ensure pep257 does not complain.

Example:

``` python
    def __init__(self, component):
        """The constructor.

        component - The component to model. Can be the name of the model or a path to a file.
        """
        self.component = component
```

I have tried searching online to see how industry treats it, but it seems almost everyone doesn't document the `__init__` unless there is a compelling reason to do so. As an example, see the [Error class in pep257](https://github.com/GreenSteam/pep257/blob/master/pep257.py#L261). :D

``` python
class Error(object):

    """Error in docstring style.

    * Stores relevant data about the error,
    * provides format for printing an error,
    * provides __lt__ method to sort errors.

    """

    # options that define how errors are printed
    explain = False
    range = False
    quote = False

    def __init__(self, filename, source, docstring, context,
                 explanation, start=None, end=None):
        self.filename = filename
        self.source = source
        self.docstring = docstring
        self.context = context
        self.explanation = explanation.strip()
```

So the question is, can/should something be added/changed in pep257 to avoid this? Is this a problem with the specification, the tool, or perhaps neither?

Any insight you guys have would be great.

**Comment by kantorii** [comment]
The following issue is still open at this point in time for this discussion.
https://github.com/PyCQA/pydocstyle/issues/189
